Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Menu/MenuManager.cpp

```diff
@@ -587,7 +610,10 @@ namespace AzToolsFramework
 
     void MenuManager::RefreshMenus()
     {
-        for (const AZStd::string& menuIdentifier : m_menusToRefresh)
+        auto menusToRefreshTemp = m_menusToRefresh;
+        m_menusToRefresh.clear();
+
+        for (const AZStd::string& menuIdentifier : menusToRefreshTemp)
```

Comment:
suggestion: A brief comment here to explain why a copy is made would be helpful for future maintainers